doc/threat_model/README.md

```diff
@@ -0,0 +1,158 @@
+Tock Threat Model
+=================
+
+## Overview
+
+Tock provides hardware-based isolation between applications as well as
+language-based isolation between untrusted kernel capsules.
+
+Tock supports a variety of hardware, including boards defined in the Tock
+repository and boards defined "out of tree" in a separate repository.
+Additionally, Tock's installation model may vary between different use cases
+even when those use cases are based on the same hardware. As a result of Tock's
+flexibility, the mechanisms it uses to provide isolation — and the strength of
+that isolation — vary from deployment to deployment.
+
+This threat model describes the isolation provided by Tock as well as the trust
+model that Tock uses to implement that isolation. Users of Tock, which include
+board integrators and application developers, should use this threat model to
+understand what isolation Tock provides to them (and what isolation it may not
+provide). Tock developers should use this threat model as a guide for how to
+provide Tock's isolation guarantees.
+
+## Definitions
+
+These definitions are shared between the documents in this directory.
+
+**Application data** includes an application's image in non-volatile storage,
+its memory footprint in RAM, and any data that conceptually belongs to the
+application that is held by the kernel or other applications. For example, if an
+application requests samples from an ADC then those samples are considered the
+application's data, even when they are stored in a location in RAM only readable
+by the kernel.
+
+**Kernel data** includes the kernel's image in non-volatile storage as well as
+data in RAM that does not conceptually belong to applications. For example, the
+scheduler's data structures are kernel data.
+
+**Capsule data** is data that is associated with a particular kernel capsule.
+This data can be either kernel data or application data, depending on its
+conceptual owner. For example, an ADC driver's sample buffer is capsule data
+associated with that ADC.
```

Comment:
Might be good to give two examples here, one for capsule data that is "kernel data" (e.g. configuration variables for a low level ADC implementation), and one for capsule data that is "application data" (e.g. the example you currently have, maybe?).

Reply:
I updated it a bit. I agree the config for a driver is generally going to be kernel data, but in retrospect a sample buffer would frequently contain application data.